src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -0,0 +1,486 @@
+                    // end_time + 1 is workaround for that
+                    taskReport.set("start_time", result.get(ZendeskConstants.Field.END_TIME).asLong() + 1);
+                }
+                else {
+                    // Sometimes no record and no end_time fetched on the job, but we should generate start_time on config_diff.
+                    taskReport.set("start_time", Instant.now().getEpochSecond());
+                }
+
+                if (numberOfRecords < ZendeskConstants.Misc.MAXIMUM_RECORDS_INCREMENTAL) {
+                    break;
+                }
+                else {
+                    startTime = result.get(ZendeskConstants.Field.END_TIME).asLong();
+                }
+            }
+        }
+        finally {
+            if (pool != null) {
+                pool.shutdown();
+                try {
+                    pool.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+                }
+                catch (final InterruptedException e) {
+                    logger.warn("Error when wait pool to finish");
+                    throw Throwables.propagate(e);
+                }
+            }
+        }
+    }
+
+    private void importDataForNonIncremental(final PluginTask task, final int taskIndex, final Schema schema,
+                                             final PageBuilder pageBuilder)
+    {
+        // Page start from 1 => page = taskIndex + 1
+        final JsonNode result = getZendeskSupportAPIService(task).getData("", taskIndex + 1, false, 0);
+        final Iterator<JsonNode> iterator = getListRecords(result, task.getTarget().getJsonName());
+
+        while (iterator.hasNext()) {
+            fetchData(iterator.next(), task, schema, pageBuilder);
+
+            if (Exec.isPreview()) {
+                break;
+            }
+        }
+    }
+
+    private Iterator<JsonNode> getListRecords(JsonNode result, String targetJsonName)
+    {
+        if (!result.has(targetJsonName) || !result.get(targetJsonName).isArray()) {
+            throw new DataException(String.format("Missing '%s' from Zendesk API response", targetJsonName));
+        }
+        return result.get(targetJsonName).elements();
+    }
+
+    private JsonNode buildColumns(final PluginTask task)
+    {
+        JsonNode jsonNode = getZendeskSupportAPIService(task).getData("", 0, true, 0);
+
+        String targetName = task.getTarget().getJsonName();
+
+        if (jsonNode.has(targetName) && jsonNode.get(targetName).isArray()) {
+            return addAllColumnsToSchema(jsonNode, task.getTarget(), task.getIncludes());
+        }
+        throw new ConfigException("Could not guess schema due to empty data set");
+    }
+
+    private final Pattern idPattern = Pattern.compile(ZendeskConstants.Regex.ID);
+    private JsonNode addAllColumnsToSchema(final JsonNode jsonNode, final Target target, final List<String> includes)
+    {
+        final JsonNode sample = createSamples(jsonNode, target);
```

Comment:
I'd suggest remove `createSamples` because the verification of empty data is similar as `buildColumns`, all we need is sub-array of 10 elements, could do via:
```
StreamSupport.stream(jsonNode.splitIterators(), false)
    .limit(10);
```